### PR TITLE
ensure alias names follows allowed pattern

### PR DESCRIPTION
**Pull request by rbayerl**
sub out unwanted characters

@grosser 
@zendesk/cloudops

### CODE REVIEW

File: lib/lambda_deployment/configuration.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@ def load_config(config_file)
     end
 
     def alias_name
-      ENV['TAG'].to_s[/\d.*/]
+      ENV['TAG'].to_s.gsub(/[^\da-zA-Z\-_]/, '')[/([0-9]+)([a-zA-Z0-9\-_]+)/]
```
Comment (grosser) [inline lib/lambda_deployment/configuration.rb]: add a comment on why this crazy regex is there ... and maybe a test for the edge-cases we discussed
